includes/amp-helper-functions.php

```diff
@@ -1854,7 +1856,7 @@ function amp_get_publisher_logo() {
 	$logo_image_url = apply_filters( 'amp_site_icon_url', $logo_image_url );
 
 	// Fallback to serving the WordPress logo.
-	if ( empty( $logo_image_url ) ) {
+	if ( empty( $logo_image_url ) && false !== $default ) {
 		$logo_image_url = amp_get_asset_url( 'images/amp-page-fallback-wordpress-publisher-logo.png' );
 	}
```

Comment:
To me this would make more sense as:

```php
if ( empty( $logo_image_url ) ) {
    $logo_image_url = ( null === $default ) ? amp_get_asset_url( 'images/amp-page-fallback-wordpress-publisher-logo.png' ) : $default;
}
```

But I think we can avoid adding the parameter altoegher. The Site Health test can simply check if the return value is equal to `amp_get_asset_url( 'images/amp-page-fallback-wordpress-publisher-logo.png' )`.